MetaMask/metamask-mobile
# fix: silence PollingBlockTracker Sentry error emit

## **Description**
1. **What is the reason for the change?**
  * Sentry PollingBlockTracker errors were very noisy (see related issue)
2. **What is the improvement/solution?**
  * Silencing the unhandled error as we weren't able to determine sources of errors
  * Patch the web3-provider-engine node module with change
  * The PollingBlockTracker will be replaced in the network-controller v9 - which is expected to be released soon

## **Related issues**

Fixes: #7579 

## **Manual testing steps**

1. Switch between networks
2. Swap some crypto
3. Send some crypto
4. Shouldn't see any PollingBlockTracker errors like in the related issue in Sentry

## **Screenshots/Recordings**

### Switch Networks

https://github.com/MetaMask/metamask-mobile/assets/141057783/c9a595cf-1ad2-4b8e-b6ec-381a2cf945ed


### Swap

https://github.com/MetaMask/metamask-mobile/assets/141057783/0daa4aff-fd15-496a-aebd-8c4d9580c23a


### Send

https://github.com/MetaMask/metamask-mobile/assets/141057783/6816e057-ce8a-43d7-abdd-5f8d74169fd4


### Transaction History

https://github.com/MetaMask/metamask-mobile/assets/141057783/bf1fe57c-bf2c-4d9e-aa4a-bbe0aab8e4bb


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
